### PR TITLE
feat(demos): demo showing WASI volume support

### LINKED ISSUE #196: Update demos to use volumes
Update all of our current demos to use volumes. This task should be done right before we release 0.2 so we don't confuse any current users of 0.1

### PR DESCRIPTION
**Pull request by bacongobbler**
closes #196

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

### CODE REVIEW

File: crates/wasi-provider/src/wasi_runtime.rs [hunk 1]
```diff
@@ -156,6 +156,11 @@ impl WasiRuntime {
 
             for (key, value) in data.dirs.iter() {
                 let guest_dir = value.as_ref().unwrap_or(key);
+                info!(
```
Comment (thomastaylor312) [inline crates/wasi-provider/src/wasi_runtime.rs:159]: This should probably be `debug!` as it is something you probably don't care about unless debugging

File: demos/wasi/hello-world-rust/k8s.yaml [hunk 2]
```diff
@@ -4,6 +4,12 @@ metadata:
   name: hello-world-wasi-rust
 data:
   myval: "cool stuff"
+  lorem_ipsum.txt: |
```
Comment (thomastaylor312) [inline demos/wasi/hello-world-rust/k8s.yaml:7]: Nit: [Bacon ipsum](https://baconipsum.com) or [hipster ipsum](https://hipsum.co) will provide more entertainment value 😆. You could also use my old haiku I wrote up

Comment (bacongobbler) [reply demos/wasi/hello-world-rust/k8s.yaml:7]: give me bacon

**Comment by bacongobbler** [comment]
good point RE: waiting until 0.2. The hello-world-rust demo is likely the first thing users experiment with. Let's hold off until 0.2 is released.

File: demos/wasi/hello-world-rust/src/main.rs [hunk 3]
```diff
@@ -8,4 +11,19 @@ fn main() {
     }
     let args: Vec<String> = env::args().collect();
     println!("Args are: {:?}", args);
+    println!("");
+
+    // open a path using the hostpath volume
+    let path = Path::new("/mnt/storage/bacon_ipsum.txt");
+    let display = path.display();
+
+    let mut file = match File::open(&path) {
```
Comment (rylev) [inline demos/wasi/hello-world-rust/src/main.rs:20]: This can be `std::fs::read_to_string`

**Comment by squillace** [comment]
@thomastaylor312 @bacongobbler I find it odd that we're concerned about people trying this out with a previous, unsupported version of Krustlet when we would document that this only works with a version that supports it -- meaning you have to compile the commit yourself. 

I thought documentation was acceptable for developers? <complete snark face asserted />

In truth, I'd merge when approved unless you specifically do not want to write documentation making it clear what works with it and what does not.

**Comment by bacongobbler** [comment]
The quickstart guide recommends installing krustlet 0.1 and using that to test. This PR pins the demonstrations on features that have not yet been released.

**Comment by squillace** [comment]
Yeah, i get it. I'd just put that in the readme: doesn't work with .1; to use this example, you must compile. Up to you.

**Comment by thomastaylor312** [comment]
I rebased and merged/pushed locally in https://github.com/deislabs/krustlet/commit/9559d47e7b2b4a5c939f4f7f7b43859f095a9940

**Pull request closed without merging**